### ISSUE TITLE
option to pass page_size filter (int value) in  constructor of google_project_alert_policies resource 

### ISSUE THREAD

**Issue by pbandari2701** [inspec/inspec-gcp#222]
Need an option to send page_size filter as part for google_project_alert_policies  resource , as the current implementation passes the page_size as nil and the response only returns policies displayed in first page of  the stack driver.

## Context
I am trying to validate different alerting policies for my project , but as the response only returns the policies from first page my expected policies are not found sometimes and the execution fails intermittently.

## Possible Implementation
pass page_size as optional value from constructor.
def initialize(opts = {})
      # Call the parent class constructor
      super(opts)
      @project = opts[:project,:page_size]
    end

and pass the same to gclient api to fetch the records based on desired page_size.
  @policies = @gcp.gcp_client(Google::Apis::MonitoringV3::MonitoringService).list_project_alert_policies("projects/#{@project}", "page_size:"@page_size)

**Comment by slevenick** [comment]
I think it would make more sense to just modify this line: https://github.com/inspec/inspec-gcp/blob/c2fae330e379c38e1c2362841a893d1cab4c2b7b/libraries/google_project_alert_policies.rb#L34

to continue calling if it returns with a page token, similar to what I did for other resources: https://github.com/inspec/inspec-gcp/blob/c2fae330e379c38e1c2362841a893d1cab4c2b7b/libraries/gcp_backend.rb#L223

Alternatively I can move this resource to be generated by Magic Modules so we get this for free

**Comment by pbandari2701** [comment]
> I think it would make more sense to just modify this line:
> 
> https://github.com/inspec/inspec-gcp/blob/c2fae330e379c38e1c2362841a893d1cab4c2b7b/libraries/google_project_alert_policies.rb#L34
> 
> to continue calling if it returns with a page token, similar to what I did for other resources:
> 
> https://github.com/inspec/inspec-gcp/blob/c2fae330e379c38e1c2362841a893d1cab4c2b7b/libraries/gcp_backend.rb#L223
> 
> Alternatively I can move this resource to be generated by Magic Modules so we get this for free

Hello @slevenick any further updates on this feature?

**Comment by slevenick** [comment]
Hey @pbandari2701 sorry for the delay.

I'm working on converting this resource to be automatically generated via Magic Modules like many of the newer resources.

Once this is done it will handle pagination

**Comment by pbandari2701** [comment]
Thanks Sam.

**Comment by pbandari2701** [comment]
Hi Sam,

Hope you are doing well.

Just wanted to inquire if you have got any forward with the magic modules
implementation for google_project_alert_policies resource. In case of
already done can you please let me know the version on inspec i should pull
this change from.

Regards,
Paramesh Bandari

**Comment by slevenick** [comment]
Hey Paramesh,

This is done and is waiting to be merged into master. Currently you can use it from master on my branch: https://github.com/modular-magician/inspec-gcp

If you want to use it prior to it being approved and merged in here, you can use this from InSpec by setting an explicit path to that repo in your inspec.yml, or cloning it and setting it to the local path. I hope to get these changes into this repo soon!

Here are the docs for the new resource: https://github.com/modular-magician/inspec-gcp/blob/master/docs/resources/google_project_alert_policies.md

For more info on inspec.yml: https://www.inspec.io/docs/reference/profiles/

**Comment by pbandari2701** [comment]
Thanks a lot Sam , i will use the reference to your branch until it is
merged into the repo.
Thanks again for this fix , i will now implement the resource in my
framework and will reach you in case of any issues i face.

Regards,
Paramesh Bandari

**Comment by slevenick** [comment]
This is merged in now
